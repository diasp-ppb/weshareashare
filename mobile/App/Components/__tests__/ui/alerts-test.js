/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CustomFormValidationMessage from '@ui/FormValidationMessage';

it('Alerts (empty) renders correctly', () => {
  const tree = renderer.create(
    <CustomFormValidationMessage />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Sucess) renders correctly', () => {
  const tree = renderer.create(
    <CustomFormValidationMessage success={'Hello Success'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Error) renders correctly', () => {
  const tree = renderer.create(
    <CustomFormValidationMessage error={'Error hey'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Status) renders correctly', () => {
  const tree = renderer.create(
    <CustomFormValidationMessage status={'Something\'s happening...'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
