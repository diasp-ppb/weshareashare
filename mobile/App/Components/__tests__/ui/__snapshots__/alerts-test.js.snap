// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts (Error) renders correctly 1`] = `
<View
  error="Error hey"
  style={
    Array [
      Object {},
      Array [
        Array [],
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#ff190c",
            "fontSize": 15,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 5,
          },
          Array [
            Object {
              "fontFamily": "HelveticaNeue",
              "marginLeft": 0,
              "marginRight": 0,
            },
            Array [],
          ],
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`Alerts (Status) renders correctly 1`] = `
<View
  status="Something's happening..."
  style={
    Array [
      Object {},
      Array [
        Array [],
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#ff190c",
            "fontSize": 15,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 5,
          },
          Array [
            Object {
              "fontFamily": "HelveticaNeue",
              "marginLeft": 0,
              "marginRight": 0,
            },
            Array [],
          ],
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`Alerts (Sucess) renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Array [],
      ],
    ]
  }
  success="Hello Success"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#ff190c",
            "fontSize": 15,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 5,
          },
          Array [
            Object {
              "fontFamily": "HelveticaNeue",
              "marginLeft": 0,
              "marginRight": 0,
            },
            Array [],
          ],
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`Alerts (empty) renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Array [],
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#ff190c",
            "fontSize": 15,
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 5,
          },
          Array [
            Object {
              "fontFamily": "HelveticaNeue",
              "marginLeft": 0,
              "marginRight": 0,
            },
            Array [],
          ],
          undefined,
        ],
      ]
    }
  />
</View>
`;
